.pre-commit-config.yaml

```diff
@@ -66,8 +66,8 @@ repos:
     hooks:
       - id: mypy
         files: ^package/|^tests/|setup.py
-        additional_dependencies: [types-setuptools==57.4.2,hypothesis==6.24.2]
-        args: [--config-file, mypy.ini]
+        additional_dependencies: [types-setuptools==57.4.2,hypothesis==6.27.0]
+        args: [--ignore-missing-imports, --config-file, mypy.ini]
```

Comment:
This config is now diverging from the [main mypy.ini](https://github.com/jenstroeger/python-package-template/blob/updates-and-tweaks/mypy.ini#L25) config?

Reply:
Hm, well, according to [this](https://github.com/pre-commit/mirrors-mypy#using-mypy-with-pre-commit):

> By default, mypy will run with `mypy --ignore-missing-imports`, `pre-commit` runs `mypy` from an isolated virtualenv so it won't have access to those. To change the arguments, override the `args` as follows:

this changes ensures both are equal, no?

Reply:
Hm, according to [mypy's documentation](https://mypy.readthedocs.io/en/latest/config_file.html#confval-ignore_missing_imports), the default value for `ignore_missing_imports` is `False` though.

Reply:
Then we’ll have to try it out to find it out 🤓

Reply:
Considering that with this change we also add
```ini
# Ignore the following imports because they have no types available.
[mypy-pytest]
ignore_missing_imports=True
```
to the `mypy.ini` file I’m tempted to remove `--ignore-missing-imports` command-line option. Just feels like duplicating settings.

Reply:
> to the mypy.ini file I’m tempted to remove --ignore-missing-imports command-line option. Just feels like duplicating settings.

I wonder if the `--ignore-missing-imports` in the [pre-commit-config.yaml](https://github.com/jenstroeger/python-package-template/blob/updates-and-tweaks/.pre-commit-config.yaml#L70) takes precedence and also applies to [mypy-src](https://github.com/jenstroeger/python-package-template/blob/updates-and-tweaks/mypy.ini#L25)?

Reply:
From [the docs](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-config-file):

> By default settings are read from `mypy.ini`, `.mypy.ini`, `pyproject.toml`, or `setup.cfg` in the current directory. Settings override mypy’s built-in defaults and command line flags can override settings.

Reply:
That's true. And that's why I suspect the pre-commit hook would apply  `ignore-missing-imports` to not only the tests by the main src modules too. We can try removing the cmd option and see if the pre-commit hook still goes through.

Reply:
I agreeably agree.